ui/ducks/metamask/metamask.js

```diff
@@ -278,10 +277,17 @@ export const getAlertEnabledness = (state) => state.metamask.alertEnabledness;
  * Get the provider configuration for the current selected network.
  *
  * @param {object} state - Redux state object.
- * @returns {import('../../../app/scripts/controllers/network/network-controller').NetworkControllerState['providerConfig']} The provider configuration for the current selected network.
  */
 export function getProviderConfig(state) {
-  return state.metamask.providerConfig;
+  const networkClientId = getSelectedNetworkClientId(state);
+  const builtInNetwork = BUILT_IN_INFURA_NETWORKS[networkClientId];
+  return builtInNetwork
+    ? {
+        ...builtInNetwork,
+        type: networkClientId,
+        rpcPrefs: { blockExplorerUrl: builtInNetwork.blockExplorerUrl },
+      }
+    : { ...getNetworkConfigurations(state)[networkClientId], type: 'rpc' };
```

Comment:
you can use `NETWORK_TYPES.RPC` instead if `'rpc'` here

Reply:
https://github.com/MetaMask/metamask-extension/pull/26150/commits/8d6d7be44fad16fd637792ffd94956a85ada6c54